### PR TITLE
add 'nanoflann' w/ example

### PR DESCRIPTION
**Pull request by headupinclouds**
original: https://github.com/jlblancoc/nanoflann
hunterized: https://github.com/hunter-packages/nanoflann

v1.2.3

`$ TOOLCHAIN=default PROJECT_DIR=examples/nanoflann ./jenkins.py --verbose`
```
Generate: 0:00:06.412748s
Build: 0:00:01.346036s
Total: 0:00:07.759096s
SUCCESS
```

### CODE REVIEW

File: cmake/projects/nanoflann/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    1.2.3-p0
+    URL
+    "https://github.com/hunter-packages/nanoflann/archive/v1.2.3-p0.tar.gz"
+    SHA1
+    72a42a31056aab380eac09b9f00dab18e91235b8
+    )
+
+hunter_pick_scheme(DEFAULT url_sha1_cmake)
+hunter_cacheable(nanoflann)
+hunter_download(PACKAGE_NAME nanoflann)
```
Comment (ruslo) [inline cmake/projects/nanoflann/hunter.cmake]: Should be the last

Comment (headupinclouds) [inline cmake/projects/nanoflann/hunter.cmake]: Done.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1770
* https://travis-ci.org/ingenue/hunter/builds/239705156?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1772
* https://travis-ci.org/ingenue/hunter/builds/239714718?utm_source=github_status&utm_medium=notification

**Comment by headupinclouds** [comment]
Thanks.

**Pull request merged**